### PR TITLE
Init kops-efs module for `efs-provider` IAM role

### PR DESCRIPTION
**Pull request by joshmyers**
## what

Module for creating an IAM role for the [efs-provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/aws/efs) to assume

## why

So that we can mount ElasticFileSystems into kops clusters.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2018 Cloud Posse, LLC
```
Comment (aknysh) [inline LICENSE]:
```suggestion
   Copyright 2019 Cloud Posse, LLC
```

File: .build-harness [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export SHELL = /bin/bash
```
Comment (aknysh) [inline .build-harness]: remove `build-harness` from Git
add `gitignore`

File: README.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# Short description of this project
+description: |-
+  Terraform module to provision an IAM role for `efs-provider` running in a Kops cluster, and attach an IAM policy to the role with permissions to mount/modify EFS.
+
+
+  ## Overview
+
+  This module assumes you are running [efs-provider](https://github.com/kubernetes-incubator/external-storage/tree/master/aws/efs) in a Kops cluster.
+
+  It will provision an IAM role with the required permissions and grant the Kops masters the permission to assume it.
```
Comment (aknysh) [inline README.yaml]: why `masters`?
or only `masters`?

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+# How to use this project
+usage: |-
+  ```hcl
+  module "kops_efs" {
+    source       = "git::https://github.com/cloudposse/terraform-aws-kops-efs.git?ref=master"
+    namespace    = "cp"
+    stage        = "prod"
+    name         = "efs-provider"
+    cluster_name = "us-east-1.cloudposse.com"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    cluster_name = "us-east-1.cloudposse.co"
```

File: README.yaml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    masters_name = "masters"
+    nodes_name   = "nodes"  
+
+    tags = {
+      Cluster = "us-east-1.cloudposse.com"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
      Cluster = "us-east-1.cloudposse.co"
```

File: outputs.tf [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  description = "IAM role unique ID"
+}
+
+output "role_arn" {
+  value       = "${aws_iam_role.default.arn}"
+  description = "IAM role ARN"
+}
+
+output "policy_name" {
+  value       = "${aws_iam_policy.default.name}"
```
Comment (aknysh) [inline outputs.tf]: `aws_iam_policy` is missing

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,56 @@
+variable "namespace" {
+  type        = "string"
+  description = "Namespace (e.g. `cp` or `cloudposse`)"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "Namespace (e.g. `eg` or `cp`)"
```

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+variable "attributes" {
+  type        = "list"
+  default     = []
+  description = "Additional attributes (e.g. `1`)"
+}
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. map(`Cluster`,`us-east-1.cloudposse.com`)"
```
Comment (aknysh) [inline variables.tf]:
```suggestion
  description = "Additional tags (e.g. map(`Cluster`,`us-east-1.cloudposse.co`)"
```

File: outputs.tf (hunk 6 above)
Comment (aknysh) [reply outputs.tf]: since `variable "policy_arn"` is provided, these three outputs are not needed

**Review by aknysh — COMMENTED** [review]
see comments

**Comment by joshmyers** [comment]
@aknysh updated

File: README.yaml [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+  The module uses [terraform-aws-kops-metadata](https://github.com/cloudposse/terraform-aws-kops-metadata) to lookup resources within a Kops cluster for easier integration with Terraform.
+
+
+# How to use this project
+usage: |-
+  ```hcl
+  module "kops_efs" {
+    source       = "git::https://github.com/cloudposse/terraform-aws-kops-efs.git?ref=master"
+    namespace    = "cp"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    namespace    = "eg"
```

**Review by aknysh — COMMENTED** [review]
see comments
rebuild README

**Review by aknysh — APPROVED** [review]

**Pull request merged**